src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,42 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import cn from 'classnames';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+interface Props {
+  selectedTodo: Todo | null;
+  closeTodo: () => void;
+}
+
+export const TodoModal: React.FC<Props> = ({ selectedTodo, closeTodo }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isError, setIsError] = useState(false);
+
+  const loadData = async () => {
+    try {
+      const currentUser = await getUser(selectedTodo?.userId || 0);
```

Comment:
You render your TodoModal component only when you have selected todo.

```suggestion
      const currentUser = await getUser(userId);
```